doc/rst/language/spec/records.rst

```diff
@@ -635,6 +635,25 @@ between the arguments' fields.  The operators ``==`` and ``!=`` check whether
 the two arguments are pair-wise equal or not.  The fields are compared in the
 order they are declared in the record definition.
 
+.. _Hashing_a_Record:
+When using a record to index into a Chapel ``map``, the compiler will generate
```

Comment:
Would be good to mention associative domains here as well, especially because ``map`` is not really covered in the language specification. Also "when using a record to index into" might be better worded. Maybe "When a record is the key for a hashtable, including when using it as the index type for an associative domain, ..."